src/v/archival/ntp_archiver_service.cc

```diff
@@ -474,7 +474,7 @@ ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
         builder.add_segments(std::move(mdiff));
         if (
           new_start_offset.has_value()
-          && old_start_offset.value_or(model::offset())
+          && old_start_offset.value_or(model::offset(0))
```

Comment:
In the case where the partition in the cloud already has start offset >0 (i.e. retention is in effect), won't this still try to truncate and get an error?

Reply:
yep, it doesn't make sense to truncate if the local manifest is empty